classes/class-gf-securesubmit.php

```diff
@@ -1,950 +1,1597 @@
+     */
+    private static $_instance = null;
+    /**
+     * @var null
+     */
+    public $transaction_response = null;
+
+    /**
+     * @return GFSecureSubmit|null
+     */
+    public static
+    function get_instance()
+    {
+        if (self::$_instance == null) {
+            self::$_instance = new GFSecureSubmit();
+        }
+
+        return self::$_instance;
+    }
+
+    /** Add our Secure CC button to the pricing fields.
+     * I found this tutorial very helpful
+     * http://wpsmith.net/2011/how-to-create-a-custom-form-field-in-gravity-forms-with-a-terms-of-service-form-field-example/
+     *
+     * @param $field_groups
+     *
+     * @return mixed
+     */
+
+
+    public
+    function hps_add_ach_field($field_groups)
+    {
+        foreach ($field_groups as &$group) {
+            if ($group["name"] == "pricing_fields") {
+                $group["fields"][] = ['class'     => 'button',
+                                      // this has to match
+                                      // \GF_Field_HPSACH::$type
+                                      'data-type' => 'hpsACH',
+                                      // the first param here will be the button text
+                                      // leave the second one as gravityforms
+                                      'value'     => __('Secure ACH', "gravityforms"),
+                                      "onclick" => "StartAddField('hpsACH');"];
+                break;
+            }
+        }
+        return $field_groups;
+    }
+
+    /**
+     *
+     */
+    public
+    function init()
+    {
+        parent::init();
+        add_action('gform_post_payment_completed',
+                   [$this,
+                    'updateAuthorizationEntry'],
+                   10,
+                   2);
+        add_filter('gform_replace_merge_tags',
+                   [$this,
+                    'replaceMergeTags'],
+                   10,
+                   7);
+        add_action('gform_admin_pre_render',
+                   [$this,
+                    'addClientSideMergeTags']);
+
+        /* Sets WP to call \GFSecureSubmit::hps_add_cc_field and build our button
+        src: wordpress/wp-includes/plugin.php
+        * \GFSecureSubmit::hps_add_cc_field
+        * * Sets WP to call \GFSecureSubmit::hps_add_ach_field and build our button
+         src: wordpress/wp-includes/plugin.php
+        * \GFSecureSubmit::hps_add_ach_field
+        */
+        add_filter('gform_add_field_buttons',
+                   [$this,
+                    'hps_add_ach_field']);
+        add_action( 'gform_editor_js_set_default_values', [$this,'set_defaults'] );
+    }
+
+    function set_defaults(){
+        ?>
+        //this hook is fired in the middle of a switch statement,
+        //so we need to add a case for our new field type
+        case "hpsACH" :
+        field.label = "Bank Transfer"; //setting the default field label
+        break;
+        <?php //////
```

Comment:
@slogsdon Yes its legit
gets added with a filter add_action('gform_editor_js_set_default_values', array($this, 'set_defaults'));
https://www.google.com/search?q=gform_editor_js_set_default_values&rlz=1C1CHBD_enUS698US698&oq=gform_editor_js_set_default_values&aqs=chrome..69i57.879j0j4&sourceid=chrome&ie=UTF-8